dsl_parser/functions.py

```diff
@@ -228,14 +226,11 @@ def validate(self, plan):
                 "get_input function references an "
                 "unknown input '{0}'.".format(input_value))
 
-    def evaluate(self, plan):
+    def evaluate(self, handler):
         if isinstance(self.input_value, list):
-            return self._get_input_attribute(plan.inputs[self.input_value[0]])
-        if self.input_value not in plan.inputs:
-            raise exceptions.UnknownInputError(
```

Comment:
Presumably this failure condition is now handled in the handler? (I'm asking this in case I forget to verify that once I hit the relevant section)

Reply:
handlers are now free to implement a missing input however they wish, from a keyerror, or a unknown input error, or hell, it can even pop an input to ask the operator